neo/framework/Common.cpp

```diff
@@ -2659,7 +2664,7 @@ void idCommonLocal::LoadGameDLL( void ) {
 	if (!gameDLL) {
 		common->Printf( "\n" );
 		common->Warning( "couldn't load mod-specific %s, defaulting to base game's library!\n", dll );
-		sys->DLL_GetFileName(BASE_GAMEDIR, dll, sizeof(dll));
+		sys->DLL_GetFileName(fs_base, dll, sizeof(dll));
 		LoadGameDLLbyName(dll, s);
 	}
```

Comment:
Maybe there should be an additional fallback: if the fs_game_base DLL (e.g. d3xp.dll) can't be loaded, fall back to the BASE_GAMEDIR DLL (base.dll)?

In that specific case that most likely won't work, but imagine a content-only addon (`foo`) based on the base game, and another content-only addon (`bar`) based on the former? Then you'd use `dhewm3 +set fs_game_base foo +set fs_game bar` but it should still use `base.dll` (because neither foo.dll nor bar.dll exists)

Reply:
Sure if you like. It crossed my mind but I decided it's not really a mechanism for stacking mods which, if they're found to work together, could be achieved by grouping the pak files in a single folder (IMHO better, in as much as it requires a conscious choice, implies compatibility, and the method is not limited to just 2).

Reply:
No idea if what I described above is actually done, but IMO it would be a much cleaner and safer way to make one mod base on another than copying the pak files together: This way several mods could base on `foo`, and you could have them installed all at once without having to worry about them breaking each other - if a mod says it needs `foo` as base mod, that means that it's compatible with it (ignoring that foo might get an update that breaks this..), and the mod could ship a .bat file to start that sets `fs_base_game foo` (like Lost Mission does for d3xp, though of course in that case you'll need d3le.dll anyway).

Furthermore, using a base mod instead of copying the paks into the same folder ensures the order in which the pk4 files are loaded (or traversed when looking for a file), making sure that when playing `bar` it first looks in its own pk4s, then in the `foo` pk4s, then in the base `pk4s` (well, it should do that at least, I hope this actually works, VFS implementations in games tend to be error-prone :-p).

Anyway, *if* there *are* any mods that are doing this right now, they *would* work without your patch (because if there's no mod-specific DLL it *always* falls back to `base.dll`), but **wouldn't** work with your patch, because then it'd suddenly demand `foo.dll` and not fall back to `base.dll`.

Reply:
It's up to you. I just think it's a very narrow case (2 content mods requiring base) which breaks down with numerous other scenarios (3+ mods, 2 content mods requiring d3xp). To my mind it's better to stick with a single methodology for mod combining (collect in one folder). Load order can be controlled by pak naming (again, a conscious choice which implies pre-tested compatibility).

Reply:
I'd like to have that fallback. It's just a few extra lines and won't break anything